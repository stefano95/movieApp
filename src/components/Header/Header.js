import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { useLocation, useNavigate, matchPath } from "react-router-dom";

import { Search, SearchIconWrapper, StyledInputBase } from "./HeaderStyles";
import { filterMovies, selectMovies } from "../../api/movieSlice";

import { Button, Grid } from "@mui/material";
import SearchIcon from "@mui/icons-material/Search";
import AddIcon from "@mui/icons-material/Add";

const Header = () => {
  const navigate = useNavigate();
  const { pathname } = useLocation();

  const dispatch = useDispatch();
  const moviesList = useSelector(selectMovies);

  const match = matchPath({ path: "/movies" }, pathname);

  const handleSearch = (e) => {
    dispatch(filterMovies(e.target.value));
  };

  return (
    <Grid
      marginX={-3}
      mb={5}
      py={2}
      gap={2}
      sx={{
        display: "flex",
        justifyContent: "center",
        flexWrap: "wrap",
        flexDirection: "column",
        backgroundColor: "rgb(200,200,200)",
      }}
    >
      {match && (
        <Grid textAlign="center" item>
          <p>
            {moviesList.length > 0
              ? `${moviesList.length} movies in
              the database`
              : "No movies found"}
          </p>
        </Grid>
      )}
      <Grid display={"flex"} justifyContent="center" flexWrap="wrap" item>
        <Grid item>
          <Search>
            <SearchIconWrapper>
              <SearchIcon />
            </SearchIconWrapper>
            <StyledInputBase
              placeholder="Searchâ€¦"
              inputProps={{ "aria-label": "search" }}
              onChange={handleSearch}
            />
          </Search>
        </Grid>
        <Grid item>
          <Button onClick={() => navigate("/movies/create")}>
            {" "}
            Add movie
            <AddIcon></AddIcon>
          </Button>
        </Grid>
      </Grid>
    </Grid>
  );
};

export default Header;
