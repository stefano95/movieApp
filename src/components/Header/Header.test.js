import React from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import { BrowserRouter, Routes, Route } from "react-router-dom";
import createStore from "../../api/store";
import { Provider } from "react-redux";
import Header from "./Header";

const store = createStore();

describe("Header", () => {
  it("renders", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Routes>
              <Route path="*" element={<Header />} />
            </Routes>
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
