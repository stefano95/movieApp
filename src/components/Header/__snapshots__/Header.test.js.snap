// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders 1`] = `
<div
  className="MuiGrid-root css-1vsh3lz-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item css-ramv3b-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <div
        className="css-1drwb0w"
      >
        <div
          className="css-1iz4iln"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </div>
        <div
          className="MuiInputBase-root MuiInputBase-colorPrimary css-zjucru-MuiInputBase-root"
          onClick={[Function]}
        >
          <input
            aria-label="search"
            className="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Searchâ€¦"
            type="text"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
         
        Add movie
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AddIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
